Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs

```diff
@@ -0,0 +1,767 @@
+                    throw new ArgumentException("No file found for: " + TransferMetadata.src_locations[0].ToString());
+                }
+            }
+
+            // Update the file metadata with GCS presigned URL
+            updateFileMetasWithPresignedUrl();
+
+            // Check command type
+            if (CommandTypes.UPLOAD == CommandType)
+            {
+                upload();
+            }
+        }
+
+        /// <summary>
+        /// Generate the result set based on the file metadata.
+        /// </summary>
+        /// <returns>The result set containing file status and info</returns>
+        public SFBaseResultSet result()
+        {
+            // Set the row count using the number of metadata in the result metas
+            TransferMetadata.rowSet = new string[ResultsMetas.Count, 8];
+
+            // For each file metadata, set the result set variables
+            for (int index = 0; index < ResultsMetas.Count; index++)
+            {
+                TransferMetadata.rowSet[index, 0] = ResultsMetas[index].srcFileName;
+                TransferMetadata.rowSet[index, 1] = ResultsMetas[index].destFileName;
+                TransferMetadata.rowSet[index, 2] = ResultsMetas[index].srcFileSize.ToString();
+                TransferMetadata.rowSet[index, 3] = ResultsMetas[index].destFileSize.ToString();
+                TransferMetadata.rowSet[index, 4] = ResultsMetas[index].resultStatus;
+
+                if (ResultsMetas[index].lastError != null)
+                {
+                    TransferMetadata.rowSet[index, 5] = ResultsMetas[index].lastError.ToString();
+                }
+                else
+                {
+                    TransferMetadata.rowSet[index, 5] = null;
+                }
+
+                if (ResultsMetas[index].sourceCompression.Name != null)
+                {
+                    TransferMetadata.rowSet[index, 6] = ResultsMetas[index].sourceCompression.Name;
+                }
+                else
+                {
+                    TransferMetadata.rowSet[index, 6] = null;
+                }
+
+                if (ResultsMetas[index].targetCompression.Name != null)
+                {
+                    TransferMetadata.rowSet[index, 7] = ResultsMetas[index].targetCompression.Name;
+                }
+                else
+                {
+                    TransferMetadata.rowSet[index, 7] = null;
+                }
+            }
+            
+            return new SFResultSet(TransferMetadata, new SFStatement(Session), externalCancellationToken);
+        }
+
+        /// <summary>
+        /// Upload files sequentially or in parallel.
+        /// </summary>
+        private void upload()
+        {
+            //Start the upload tasks(for small files upload in parallel using the given parallelism 
+            //factor, for large file updload sequentially)
+            //For each file, using the remote client
+            if (0 < LargeFilesMetas.Count)
```

Comment:
It will check if overwrite is enabled: https://github.com/snowflakedb/snowflake-connector-net/pull/415/files#diff-141fae8d9ec0879939309639fad147fd50b1bba1902c0dbd13fd25fb5c10e3c7R108

If overwrite is disabled and file already exists, then it will skip that file
If overwrite is enabled, then it will re-upload the file